src/app.tsx

```diff
@@ -1,14 +1,30 @@
-import React from 'react';
+import React, { useState } from 'react';
 
 import { Router } from 'routes';
 import { routeConfig } from 'route-components';
 import 'index.scss';
 import { ErrorBoundary } from 'common/error-boundary';
+import { UserContext } from './common/user-context';
 
-const App = () => (
-  <ErrorBoundary>
-    <Router routeConfig={routeConfig} />
-  </ErrorBoundary>
-);
+const App = () => {
+  const [filteredUsers, setFilteredUsers] = useState<DummyUser[]>([]);
+  const [unfilteredUsers, setUnfilteredUsers] = useState<DummyUser[]>([]);
+
+  // eslint-disable-next-line react/jsx-no-constructed-context-values
+  const value = {
+    unfilteredUsers,
+    filteredUsers,
+    updateUnfilteredUsers: setUnfilteredUsers,
+    updateFilteredUsers: setFilteredUsers,
+  };
+
+  return (
+    <UserContext.Provider value={value}>
```

Comment:
I would say we want the `UserContext` to still be wrapped by the `ErrorBoundary`. That way, if there is an unhandled exception inside the `UserContext`, it will be caught by the `ErrorBoundary`.